app/lib/actions/middleware/watch_delegated_proxy_sub_tasks.rb

```diff
@@ -7,13 +7,12 @@ class CheckOnProxyActions; end
       BATCH_SIZE = 1000
 
       def run(event = nil)
-        if event.nil?
-          set_clock
-        elsif event == CheckOnProxyActions
+        if event == CheckOnProxyActions
           check_triggered
           set_clock
           action.send(:suspend)
         end
+        set_clock
```

Comment:
Shouldn't we still set the clock just at the first run (with event.nil?) and then only on `CheckOnProxyActions`. Why is this change needed?